### PR TITLE
DEX fixes

### PR DESCRIPTION
**Pull request by ukane-philemon**
- Prevent manual bond posting if dex client is maintaining the tier
- Improve the cancel order button
- Don't allow cancel for active orders

### CODE REVIEW

File: ui/page/dcrdex/market.go [hunk 1]
```diff
@@ -1776,14 +1794,13 @@ func (pg *DEXMarketPage) refreshOrders() {
...
+		if ord.Status == order.OrderStatusExecuted && hasActiveMatch != pg.openOrdersDisplayed /* display active orders on open order view */ {
 			continue // skip order
 		}
 
-		if pg.openOrdersDisplayed && !ord.Cancelling {
-			btn := pg.Theme.DangerButton(values.String(values.StrCancel))
-			btn.Margin = layout.Inset{}
-			ord.cancelBtn = &btn
+		if pg.openOrdersDisplayed && !ord.Cancelling && !hasActiveMatch {
+			ord.cancelBtn = pg.Theme.NewClickable(false)
```
Comment (itswisdomagain) [inline ui/page/dcrdex/market.go:1803]: You can cancel even if there's an active match; if there's an unmatched quantity. I think the correct check would be `if ord.Status != order.OrderStatusExecuted`.

**Review by itswisdomagain — CHANGES_REQUESTED** [review]

File: ui/page/dcrdex/market.go [hunk 2]
```diff
@@ -721,14 +721,32 @@ func (pg *DEXMarketPage) orderForm(gtx C) D {
...
+			setting, err := dexClient.WalletSettings(bondAssetID)
+			if err != nil {
+				pg.notifyError(err.Error())
+			} else {
+				// Wallet is been used by the dex client so it exists, can ignore errors.
+				walletID, _ := strconv.Atoi(setting[dexc.WalletIDConfigKey])
+				accountNumber, _ := strconv.Atoi(setting[dexc.WalletAccountNumberConfigKey])
+				asset := pg.AssetsManager.WalletWithID(walletID)
+				accountName, _ := asset.AccountName(int32(accountNumber))
+				bondAmtString := calculateBondAmount(asset, xc.BondAssets[convertAssetIDToAssetType(bondAssetID).ToStringLower()], int(targetTier), dexClient.BondsFeeBuffer(bondAssetID))
```
Comment (itswisdomagain) [inline ui/page/dcrdex/market.go:744]:
```suggestion
				bondAmtString := calculateBondAmount(asset, xc.BondAssets[asset.GetAssetType().ToStringLower()], int(targetTier), dexClient.BondsFeeBuffer(bondAssetID))
```

File: ui/page/dcrdex/market.go [hunk 3]
```diff
@@ -721,14 +721,32 @@ func (pg *DEXMarketPage) orderForm(gtx C) D {
...
 		overlayMsg = values.String(values.StrPostBondMsg)
-		actionBtn = &pg.postBondBtn
+		targetTier := xc.Auth.TargetTier
+		if targetTier > 0 { // Maintenance enabled
+			bondAssetID := xc.Auth.BondAssetID
+			setting, err := dexClient.WalletSettings(bondAssetID)
+			if err != nil {
+				pg.notifyError(err.Error())
+			} else {
+				// Wallet is been used by the dex client so it exists, can ignore errors.
```
Comment (itswisdomagain) [inline ui/page/dcrdex/market.go:739]:
```suggestion
				// Wallet is being used by the dex client so it exists, can ignore errors.
```

File: ui/page/dcrdex/market.go [hunk 4]
```diff
@@ -1286,7 +1304,7 @@ func (pg *DEXMarketPage) orderColumn(header bool, txt string, columnWidth unit.D
 				lb.Color = pg.Theme.Color.Text
 				return lb.Layout(gtx)
 			} else if showCancelBtn {
-				return pg.orders[orderIndex].cancelBtn.Layout(gtx)
+				return pg.orders[orderIndex].cancelBtn.Layout(gtx, pg.Theme.Icons.FailedIcon.Layout24dp)
```
Comment (itswisdomagain) [inline ui/page/dcrdex/market.go:1307]: FailedIcon?

File: ui/page/dcrdex/market.go [hunk 5]
```diff
@@ -1796,14 +1812,13 @@ func (pg *DEXMarketPage) refreshOrders() {
...
-			// displayed first.
-			sort.SliceStable(orders, func(i, j int) bool {
-				return pg.orders[i].SubmitTime > orders[j].SubmitTime
-			})
 		}
 	}
+
+	// Always sort orders.
+	sort.SliceStable(pg.orders, func(i, j int) bool {
+		return pg.orders[i].SubmitTime > orders[j].SubmitTime
```
Comment (itswisdomagain) [inline ui/page/dcrdex/market.go:1820]: Had a out-of-range panic here once.
```suggestion
		return pg.orders[i].SubmitTime > pg.orders[j].SubmitTime
```

**Review by itswisdomagain — CHANGES_REQUESTED** [review]

File: ui/page/dcrdex/market.go (hunk 4 above)
Comment (ukane-philemon) [reply ui/page/dcrdex/market.go:1307]: it's the "x" wit red bg icon.

Comment (ukane-philemon) [reply ui/page/dcrdex/market.go:1307]: it's the "x" wit red bg icon.

**Review by itswisdomagain — APPROVED** [review]

**Pull request merged**